src/webgpu/api/operation/vertex_state/index_format.spec.ts

```diff
@@ -0,0 +1,380 @@
+export const description = `
+Test indexing, index format and primitive restart.
+`;
+
+import { params, poptions } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { getTextureCopyLayout } from '../../../util/texture/layout.js';
+
+const kHeight = 4;
+const kWidth = 8;
```

Comment:
I expanded the 4x4 image by 4 more pixels on the left, and moved index -1 over there, so that it would render things.

Mostly this just meant adding space to the left of the expected results and then adding extra triangles on top of the old expected results.
I noticed that some of the triangles cut right along pixel centers, so hopefully rounding issues don't make this differ on different hardware. We'll want to update the test if we encounter that.